src/main/java/MongoDBDAO.java

```diff
@@ -1,139 +1,140 @@
+    public String setWeatherstationData(String jsonMessage, String weatherstationName) throws ServerException {
+        //For Production:
+        //MongoCollection<Document> collection = MongoConnection.getInstance("AgricircleDB").getDatabase().getCollection(weatherstationName);
 
+        //For Development:
+        MongoCollection<Document> collection = MongoConnection.getInstance("AgricircleDB").getDatabase().getCollection("Weatherstation1");
         try {
             collection.insertOne(Document.parse(jsonMessage));
             return "Database updated";
-        } catch (MongoException mwe) {
-            //  Block of code to handle error
-            return mwe.getMessage();
-            //throw new DBException("Soe")
+        } catch (MongoWriteException | MongoWriteConcernException e) {
+            throw new ServerException(e.getMessage());
         }
     }
 
     @Override
-    public String findAllFromDatabase() {
+    public String getWeatherstationData(String weatherstationName , String contentType)
+    {
+        //For Production:
+        //MongoCollection<Document> collection = MongoConnection.getInstance("AgricircleDB").getDatabase().getCollection(weatherstationName);
 
+        //For Development:
         MongoCollection<Document> collection = MongoConnection.getInstance("AgricircleDB").getDatabase().getCollection("Weatherstation1");
-
-        try {
-            StringBuilder returnString = new StringBuilder();
-            FindIterable<Document> iterable = collection.find();
+        StringBuilder returnString = new StringBuilder();
+        FindIterable<Document> iterable = collection.find().projection(fields(include(
+                "payload_fields.avg_wind_speed",
+                "payload_fields.solar_radiation",
+                "payload_fields.outside_temperature",
+                "payload_fields.outside_humidity",
+                "payload_fields.barometer_data",
+                "payload_fields.rain_rate"),
+                excludeId()));
+        if(contentType.equals("netcdf"))
+        {
+            //TODO Convert JSON to netcdf and return it.
+            return "Not Yet Implemented. Use contenttype = json";
+        }
+        if(iterable != null) {
+            returnString.append("[");
             for (Document document : iterable) {
-                returnString.append(document.toJson());
+                returnString.append(document.toJson()).append(",");
             }
-            return returnString.toString();
-        } catch (MongoException mwe) {
-            //  Block of code to handle errors
-            return mwe.getMessage();
+            returnString.delete(returnString.length()-1 , returnString.length());
+            returnString.append("]");
         }
+        if(returnString.length() == 0)
+        {
+            returnString.append("We couldn't find any data with the given criteria");
+        }
+        return returnString.toString();
```

Comment:
Redundant toString on String

Reply:
returnString is a StringBuilder object, which is the reason i'm using toString on it.